packages/sdk/src/transactions/OrdTransaction.ts

```diff
@@ -229,26 +226,29 @@ export class OrdTransaction {
       redeem: redeemScript
     })
 
-    // inscription tx always have 1 input and 1 + n custom outs
-    const fees = calculateTxFee({
-      totalInputs: 1,
-      totalOutputs: 1 + this.#outs.length,
-      satsPerByte: this.feeRate,
-      type: "taproot", // hardcoding because this process is only supported by Taproot txs
-      additional: { witnessScripts: inscribePayTx.witness }
+    this.#suitableUnspent = getDummyP2TRInput()
+    this.build()
+    const feeEstimator = new FeeEstimator({
+      psbt: this.psbt!,
+      feeRate: this.feeRate,
+      network: this.network
     })
+    const fee = feeEstimator.calculateNetworkFee()
+    this.psbt = null
+    this.#suitableUnspent = null
+    this.inputsToSign.signingIndexes.pop() // remove last added index
```

Comment:
I think having a pure function for `this.build()` is better than setting the dummy unspents and having to set it back to null later. I am not sure what side-effect might be caused if there is error in between the code

so purefunctions like `this.buildPSBT({...}` that takes in all the params needed to build the PSBT and returns the PSBT, so no external data is being modified in the function.

Reply:
Too many moving parts, I am not sure if creating a pura fn is a good idea. However, we can revisit this later if the current implementation becomes an issue.